### ISSUE TITLE
Flag to turn reaction kineticLaw parameters into model parameters

### ISSUE THREAD

**Issue by oxinabox** [sebapersson/SBMLImporter.jl#62]
I was comparing this to MEWpy, and in MEWpy all the parameters of reactions' kinetic laws end up as parameters of the model.
But in systems imported with SBMLImporter, all of those are inlined directly as constants.
I think it happens here: https://github.com/sebapersson/SBMLImporter.jl/blob/main/src/reactions.jl#L63

I propose a flag that instead of inlining them makes them parameters of the model itself, and push them into the `p` vector
So I can modify them from MTK.

One possible problem is their `ids` may clash between reactions?
in the model i am looking at they don't so maybe SBML standard forbids that (?).
But I guess can error in that case (unless they have same value as well? in which case merge them?)

For example the following: has a bunch of literals in it:
```
julia> vPTS.rate
(7829.78extracellular*cglcex(t)*cpep(t)) / ((1 + 0.46511627906976744(cg6p(t)^3.66))*(3082.3 + 245.3cglcex(t) + (0.01cpep(t)) / cpyr(t) + (cglcex(t)*cpep(t)) / cpyr(t))*cpyr(t))
```
but I would want to see instead:

```
(rmaxPTS*extracellular*cglcex(t)*cpep(t)) / ((1 + (1/KPTSg6p)*(cg6p(t)^nPTSg6p))*(KPTSa1 + KPTSa3*cglcex(t) + (KPTSa2*cpep(t)) / cpyr(t) + (cglcex(t)*cpep(t)) / cpyr(t))*cpyr(t))
```

Source XML:
```xml
 <reaction metaid="_704943" id="vPTS" name="Phosphotransferase system" reversible="false">
        <annotation>
          <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dcterms="http://purl.org/dc/terms/" xmlns:vCard="http://www.w3.org/2001/vcard-rdf/3.0#" xmlns:bqbiol="http://biomodels.net/biology-qualifiers/" xmlns:bqmodel="http://biomodels.net/model-qualifiers/">
            <rdf:Description rdf:about="#_704943">
              <bqbiol:is>
                <rdf:Bag>
                  <rdf:li rdf:resource="urn:miriam:kegg.pathway:eco02060"/>
                </rdf:Bag>
              </bqbiol:is>
            </rdf:Description>
          </rdf:RDF>
        </annotation>
        <listOfReactants>
          <speciesReference species="cglcex"/>
          <speciesReference species="cpep"/>
        </listOfReactants>
        <listOfProducts>
          <speciesReference species="cg6p"/>
          <speciesReference species="cpyr"/>
        </listOfProducts>
        <kineticLaw>
          <math xmlns="http://www.w3.org/1998/Math/MathML">
            <apply>
              <divide/>
              <apply>
                <times/>
                <ci> extracellular </ci>
                <ci> rmaxPTS </ci>
                <ci> cglcex </ci>
                <apply>
                  <divide/>
                  <ci> cpep </ci>
                  <ci> cpyr </ci>
                </apply>
              </apply>
              <apply>
                <times/>
                <apply>
                  <plus/>
                  <ci> KPTSa1 </ci>
                  <apply>
                    <times/>
                    <ci> KPTSa2 </ci>
                    <apply>
                      <divide/>
                      <ci> cpep </ci>
                      <ci> cpyr </ci>
                    </apply>
                  </apply>
                  <apply>
                    <times/>
                    <ci> KPTSa3 </ci>
                    <ci> cglcex </ci>
                  </apply>
                  <apply>
                    <times/>
                    <ci> cglcex </ci>
                    <apply>
                      <divide/>
                      <ci> cpep </ci>
                      <ci> cpyr </ci>
                    </apply>
                  </apply>
                </apply>
                <apply>
                  <plus/>
                  <cn type="integer"> 1 </cn>
                  <apply>
                    <divide/>
                    <apply>
                      <power/>
                      <ci> cg6p </ci>
                      <ci> nPTSg6p </ci>
                    </apply>
                    <ci> KPTSg6p </ci>
                  </apply>
                </apply>
              </apply>
            </apply>
          </math>
          <listOfParameters>
            <parameter id="rmaxPTS" value="7829.78" units="mM_per_second"/>
            <parameter id="KPTSa1" value="3082.3" units="mM"/>
            <parameter id="KPTSa2" value="0.01" units="mM"/>
            <parameter id="KPTSa3" value="245.3" units="dimensionless"/>
            <parameter id="nPTSg6p" value="3.66" units="dimensionless"/>
            <parameter id="KPTSg6p" value="2.15" units="mM"/>
          </listOfParameters>
        </kineticLaw>
      </reaction>
      ```

**Comment by oxinabox** [comment]
The file I am looking at is
https://raw.githack.com/BioSystemsUM/MEWpy/6a5031d6b8c654e15b7b53778d7a8186ad735b23/examples/models/kinetic/chassagnole2002.xml

But it is literally the second SBML file I have looked at ever.
(and the first was Yeast9, so much bigger I think)

**Comment by sebapersson** [comment]
I can work with this file :) and also put together another smaller test case using one of the models we already test on.

Unless I run into something fully unexpected while implementing this I should have the features by latest Thursday

**Comment by sebapersson** [comment]
This is now fixed by #64. As can now be found in the dev [API](https://sebapersson.github.io/SBMLImporter.jl/dev/API) it is possible to set `inline_kineticlaw_parameters=false` as an argument to `load_SBML`.

Adding this functionality proved to be a bit more subtle than I had expected, as for kinetic law parameters (parameters only defined in the SBML reactions) there might be duplicate IDs. I decided to use the following heuristics to handle this:
1. If there is a duplicate id, but that id defines parameters with identical values (and other fields) assume the duplicate ids refers to the same parameter.
2. If there are duplicate ids that defines parameters with different values throw an error.

As soon as all tests are done I will make a minor release where the features from here and #61 are available.
